### PR TITLE
Migrate antlr and base-services builds to Kotlin

### PR DESCRIPTION
**Pull request by bamboo**
This PR provides code review comments as commits.

### CODE REVIEW

**Comment by eskatos** [comment]
Feedback build: https://builds.gradle.org/viewQueued.html?itemId=10993038

**Comment by eskatos** [comment]
New feedback build: https://builds.gradle.org/viewQueued.html?itemId=11000953

File: subprojects/base-services/base-services.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        setProperty("include", listOf("HashingAlgorithmsBenchmark"))
+    }
+}
+
+val generatedTestResourcesDir: File by extra
+
+val buildReceiptPackage: String by rootProject.extra
+
+val buildReceiptResource by tasks.creating(Copy::class) {
+    from(Callable { tasks.getByPath(":createBuildReceipt").outputs.files })
```
Comment (JLLeitschuh) [inline subprojects/base-services/base-services.gradle.kts]: :o: I still find this API on the copy task incredibly clunky and totally unintuitive as a user.

File: subprojects/base-services/base-services.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    destinationDir = file("$generatedTestResourcesDir/$buildReceiptPackage")
+}
+
+java.sourceSets {
+    "main" {
+        output.dir(mapOf("builtBy" to buildReceiptResource), generatedTestResourcesDir)
```
Comment (JLLeitschuh) [inline subprojects/base-services/base-services.gradle.kts]: Is there any plan to add a concrete class around the idea of `builtBy` for artifact declarations and cases like this?

Comment (bamboo) [reply subprojects/base-services/base-services.gradle.kts]: There's a short term plan of adding proper overloads to allow named argument syntax, e.g. `output.dir(builtBy = buildReceiptResource, ...)`.

**Pull request merged**